cli/src/config.rs

```diff
@@ -229,20 +231,48 @@ impl UnresolvedConfigEnv {
             config_dir.push("conf.d");
             ConfigPath::new(config_dir)
         });
+
         use ConfigPath::*;
-        if let Some(path @ Existing(_)) = home_config_path {
-            paths.push(path);
-        } else if let (Some(path @ New(_)), None) = (home_config_path, &platform_config_path) {
-            paths.push(path);
+
+        match home_config_path {
+            Some(path @ Existing(_)) => paths.push(path),
+            Some(path @ New(_)) if platform_config_path.is_none() => paths.push(path),
+            _ => (),
+        }
+
+        #[cfg(target_os = "macos")]
```

Comment:
nit: missing TODO to remove after jj 0.35+, since it will no longer be supported then.